doseid/src/server/integration/mod.rs

```diff
@@ -1 +1,12 @@
+use git2::Repository;
+use std::path::Path;
 pub(crate) mod github;
+
+// TODO: Design concept of Git Source Integration trait
+pub trait GitSource {
```

Comment:
This is a welcome addition. Given that `BitBucket`, `Github` and `Gitlab` will have to conform to this, makes a good case of being a trait.